docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+
+There are two strategies to avoid this clash:
+1. We can store each of these values in their own key and only hold the key at which they are stored in the `task1` object itself. This would mean your `SharedMap` could have an object like:
+
+```json
+{
+    "task1": {
+        "titleKey": "task1Title",
+        "descriptionKey": "task1Description",
+        "assignedTo": "task1AssignedTo"
+    },
+    "task1Title": "Awesome Task",
+    "task1Description": "Doing the most awesome things",
+    "task1AssignedTo": {
+        "name": "Joe Schmo",
+        "email": "joeschmo@email.com",
+        "address": "1234 Fluid Way"
+    }
+}
+```
+
+With this, when user A is editing the title, they could be executing the following code:
+
+```javascript
+const task = map.get("task1");
+map.set(task.titleKey, editedTitle)
+```
+
+Similarly, when user B is editing the description, they could be executing the following:
+
+```javascript
+const task = map.get("task1");
+map.set(task.descriptionKey, editedDescription)
+```
+
+Now, each user would be updating the fields independently and one would not be overwriting the other, as the parent `task1` object isn't the one that is being set each time but rather just holding the pointers to each of the separate fields being edited.
+
+2. One of the caveats of the above approach is that both the tasks as well as their values are now all stored at the same level within the map. I.e. if you call `map.values()`, it will provide both the tasks themselves as well as each of their individual fields. Instead, you can have each task be stored in its own `SharedMap` and have a parent `SharedMap` that keeps track of all of the different tasks under it. We will take a look at this in the **Nested shared objects example** section below.
+
+Now, you may be asking if you would need to do the same for `assignedTo` object. The answer is "yes" if the fields in this object were also something you would be expecting users modularly edit. However, if this object will always be set together then you can keep it under the same key. Since this user metadata is coming as a singular blob from a service, you can use the latter in this scenario.
+
+```javascript
+// Pseudocode for a function that makes a backend call to fetch a roster of all users in an array
+const users = await getAvailableUsers();
+
+// Let's say that the user the task is assigned to is at index pickedUserIndex
+const pickedUser = users[pickedUserIndex];
+
+// Now store this user object as a whole into the SharedMap
+const task = map.get("task1");
+map.set(task.assignedToKey, pickedUser)
+```
+
+This will work as expected **because the entire object is being stored each time** instead of specific fields.
+
+One way to think about this is that each value stored into the `SharedMap` is a unit of collaboration. Any time you want users to be able to collaborate on invidual pieces of an object, it should be appropriately stored in separate keys.
```

Comment:
```suggestion
This will work as expected **because the entire object is being stored each time** instead of specific fields.

One way to think about this is that each value stored into the `SharedMap` is the smallest simultaneously editable piece of data. Any time you want users to be able to simultaneously edit individual pieces of an object, you should store those properties in separate keys.
```